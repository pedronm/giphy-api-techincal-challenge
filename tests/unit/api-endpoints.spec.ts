import Axios from "axios";

describe("GIPHY API: Most Popular", () => {
    const requestURL = `${process.env.VUE_APP_GIPHYAPI_MOST_POPULAR}?api_key=${process.env.VUE_APP_GIPHYAPI_KEY}&limit=25&rating=G`;    
    it("response image data has not changed", () => {
        return Axios.get(requestURL).then(res => {
            const imageData = res.data.data
            const sample = imageData[0]
            const keys = Object.keys(sample);
            const formats = Object.keys(sample.images);
            expect(keys).toMatchSnapshot();
            expect(formats).toMatchSnapshot();
        })
    });  
});