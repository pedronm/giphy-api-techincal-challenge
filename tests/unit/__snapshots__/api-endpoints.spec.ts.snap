// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GIPHY API: Most Popular response image data has not changed 1`] = `
Array [
  "type",
  "id",
  "url",
  "slug",
  "bitly_gif_url",
  "bitly_url",
  "embed_url",
  "username",
  "source",
  "title",
  "rating",
  "content_url",
  "source_tld",
  "source_post_url",
  "is_sticker",
  "import_datetime",
  "trending_datetime",
  "images",
  "user",
  "analytics_response_payload",
  "analytics",
]
`;

exports[`GIPHY API: Most Popular response image data has not changed 2`] = `
Array [
  "hd",
  "downsized_large",
  "fixed_height_small_still",
  "original",
  "fixed_height_downsampled",
  "downsized_still",
  "fixed_height_still",
  "downsized_medium",
  "downsized",
  "preview_webp",
  "original_mp4",
  "fixed_height_small",
  "fixed_height",
  "downsized_small",
  "preview",
  "fixed_width_downsampled",
  "fixed_width_small_still",
  "fixed_width_small",
  "original_still",
  "fixed_width_still",
  "looping",
  "fixed_width",
  "preview_gif",
  "480w_still",
]
`;
